src/main/java/com/ncc/aif/AIFUtils.java

```diff
@@ -0,0 +1,716 @@
+                                           List<Double> vectorData, Resource system) throws JsonProcessingException {
+        final ObjectMapper mapper = new ObjectMapper();
+        final ImmutableMap<String, Object> jsonMap =
+                ImmutableMap.of("vector_type", vectorType, "vector_data", vectorData);
+        return markPrivateData(model, resource, mapper.writeValueAsString(jsonMap), system);
+    }
+
+    public static Resource linkToExternalKB(Model model, Resource toLink, String externalKbId, Resource system,
+                                            Double confidence) {
+        final Resource linkAssertion = makeAIFResource(model, null, AidaAnnotationOntology.LINK_ASSERTION_CLASS, system);
+        toLink.addProperty(AidaAnnotationOntology.LINK, linkAssertion);
+        linkAssertion.addProperty(AidaAnnotationOntology.LINK_TARGET, model.createTypedLiteral(externalKbId));
+        if (confidence != null) {
+            markConfidence(model, linkAssertion, confidence, system);
+        }
+        return linkAssertion;
+    }
+
+
+    static final class SparqlQueries {
+        static final Query TYPE_QUERY = QueryFactory.create(
+                ("PREFIX rdf: <" + RDF.uri + ">\n" +
+                        "SELECT ?typeAssertion WHERE {\n" +
+                        "?typeAssertion a rdf:Statement .\n" +
+                        "?typeAssertion rdf:predicate rdf:type .\n" +
+                        "?typeAssertion rdf:subject ?typedObject .\n" +
+                        "}").replace("\n", System.getProperty("line.separator")));
+    }
+
+    private static Resource makeAIFResource(Model model, String uri, Resource classType, Resource system) {
+        Resource resource = (uri == null ? model.createResource() : model.createResource(uri));
+        resource.addProperty(RDF.type, classType);
+        markSystem(resource, system);
+        return resource;
+    }
+}
+
+/**
+ * A strategy for generating RDF graph nodes
+ */
+interface IriGenerator {
+    String nextIri();
+}
+
+/**
+ * A node generation strategy which uses UUIDs appended to a base URI.
+ */
+final class UuidIriGenerator implements IriGenerator {
+    private String baseUri;
+
+    UuidIriGenerator() {
+        baseUri = "dummy.uri";
+    }
+
+    UuidIriGenerator(String baseUri) {
+        if (baseUri == null || baseUri.isEmpty()) {
+            throw new IllegalArgumentException("Base URI cannot be empty");
+        }
+        if (!baseUri.substring(1).contains(":")) {
+            throw new IllegalArgumentException("Base URI must contain a prefix followed by a colon separator");
+        }
+        if (baseUri.endsWith("/")) {
+            throw new IllegalArgumentException("Base URI cannot end in /");
+        }
+        this.baseUri = baseUri;
+    }
+
+    @Override
+    public String nextIri() {
+        return baseUri + '/' + UUID.randomUUID().toString();
+    }
+}
```

Comment:
Sorry, I should have caught this on the first pass. The old-school Linux heads have told me that every line needs an endline character. Personally, I think it's an out-dated requirement, but you should probably add them for conformity.